Framework/API/test/WorkspaceListPropertyTest.h

```diff
@@ -0,0 +1,293 @@
+  };
+
+  /**
+  * Helper class. Algorithms are instances of IPropertyManager
+  */
+  class MyAlgorithm : public Mantid::API::Algorithm {
+  public:
+    MyAlgorithm() { this->setRethrows(true); }
+
+    virtual int version() const { return 1; }
+
+    virtual const std::string name() const { return "MyAlgorithm"; }
+
+    const std::string summary() const override { return "MyAlgorithm helper."; }
+
+    virtual void init() {
+      declareProperty(std::make_unique<WorkspaceListProperty<Workspace>>(
+          "MyProperty", std::vector<Workspace_sptr>(0)));
+    }
+
+    virtual void exec() {
+      std::vector<Workspace_sptr> val = getProperty("MyProperty");
+    }
+
+    virtual ~MyAlgorithm() {}
+  };
+
+  /**
+  * Helper class. Algorithms are instances of IPropertyManager
+  */
+  class MyAlgorithm2 : public Mantid::API::Algorithm {
+  public:
+    MyAlgorithm2() { this->setRethrows(true); }
+
+    virtual int version() const { return 1; }
+
+    virtual const std::string name() const { return "MyAlgorithm"; }
+
+    const std::string summary() const override { return "MyAlgorithm helper."; }
+
+    virtual void init() {
+      declareProperty(std::make_unique<WorkspaceListProperty<MatrixWorkspace>>(
+          "MyProperty", std::vector<MatrixWorkspace_sptr>(0)));
+    }
+
+    virtual void exec() {
+      std::vector<MatrixWorkspace_sptr> val = getProperty("MyProperty");
+    }
+
+    virtual ~MyAlgorithm2() {}
+  };
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static WorkspaceListPropertyTest *createSuite() {
+    return new WorkspaceListPropertyTest();
+  }
+  static void destroySuite(WorkspaceListPropertyTest *suite) { delete suite; }
+
+  //------------------------------------------------------------------------------
+  // Functional Testing
+  //------------------------------------------------------------------------------
+
+  void test_constructor() {
+    auto list = createWorkspaceList();
+
+    TS_ASSERT_THROWS_NOTHING(std::make_unique<WorkspaceListProperty<Workspace>>(
+        WorkspaceListProperty<Workspace>("MyWorkspaceProperty", list,
+                                         Direction::Input,
+                                         PropertyMode::Mandatory)));
+  }
+
+  void test_contruct_as_optional() {
+    auto list = std::vector<Workspace_sptr>();
+
+    TS_ASSERT_THROWS_NOTHING(std::make_unique<WorkspaceListProperty<Workspace>>(
+        WorkspaceListProperty<Workspace>("MyWorkspaceProperty", list,
+                                         Direction::Input,
+                                         PropertyMode::Optional)));
+  }
+
+  void test_copy_construction() {
+    auto list = createWorkspaceList();
+
+    WorkspaceListProperty<Workspace> a("PropA", list, Direction::Input,
+                                       PropertyMode::Optional);
+    WorkspaceListProperty<Workspace> b(a);
```

Comment:
If `WorkspaceListProperty<T>` does not overload `operator==` it might be a good idea to add it so that you can use `TS_ASSERT_EQ` directly in the tests rather than re-imagining the equality operator here.